### PR TITLE
CI build wheels for macosx

### LINKED ISSUE #268: Add support for x86_64 macos wheels
## ✨ Feature Request

There is community appetite for `cf-units` wheels on MacOS `x86_64`, reference #258.

Contact @alexgleith.

### PR DESCRIPTION
**Pull request by bjlittle**
### Description
This PR extends the GHA CI wheel building to `macosx`.

Massive thanks to @alexgleith for super responsive testing of the `macosx` wheels on the fly :partying_face: 

See #258

Closes #268

### CODE REVIEW

File: .github/workflows/ci-wheels.yml [hunk 1]
```diff
@@ -26,35 +26,49 @@ concurrency:
...
 jobs:
   build_bdist:
-    name: "Build bdist ${{ matrix.arch }} ${{ matrix.os }}"
-    runs-on: ${{ matrix.os }}
+    name: "Build ${{ matrix.os }} (${{ matrix.arch }}) wheels"
+    runs-on: ${{ matrix.os }}-latest
     strategy:
       fail-fast: false
       matrix:
-        # TBD: extend to support osx and windows
```
Comment (trexfeathers) [inline .github/workflows/ci-wheels.yml:34]: Do we no longer have an ambition to support Windows?

File: MANIFEST.in [hunk 2]
```diff
@@ -1,9 +1,13 @@
...
-include *.md
-recursive-include cf_units *.py *.pxd *.pyx *.c
-include cf_units/etc/site.cfg.template
-include CHANGES COPYING COPYING.LESSER INSTALL
-exclude cf_units/etc/site.cfg
-graft requirements
+# Top-level files
+include CHANGES COPYING COPYING.LESSER INSTALL README.md
 prune .github
-exclude cf_units/_version.py
```
Comment (trexfeathers) [inline MANIFEST.in:9]: Why is this different to SciTools/iris#4849, where we continue to exclude `_version.py`?

**Review by trexfeathers — COMMENTED** [review]
Clever bit of flexibility 🙂
Couple of questions for you

File: MANIFEST.in (hunk 2 above)
Comment (bjlittle) [reply MANIFEST.in:9]: Actually, after thinking this through and experimenting, we **always** want the `_version.py` to be included in an sdist or bdist to guarantee that the version is correct when there is no git history available to `setuptools-scm`.

Although, it still stands the the `_version.py` **shouldn't** be put under version control.

I'll be aligning `iris` with this change too :+1:

File: .github/workflows/ci-wheels.yml (hunk 1 above)
Comment (bjlittle) [reply .github/workflows/ci-wheels.yml:34]: I'll add the comment back, but only have a `TBD` for `windows` :+1:

**Review by trexfeathers — APPROVED** [review]
Thanks @bjlittle

**Comment by bjlittle** [comment]
@trexfeathers Awesome, thanks for the swift review :100: :rocket:

**Pull request merged**